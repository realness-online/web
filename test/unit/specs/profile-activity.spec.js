import Vue from 'vue'
import ProfileActivity from '@/components/profile-activity'

describe('profile-activity.vue', () => {
  let Constructor

  beforeEach(() => {
    Constructor = Vue.extend(ProfileActivity)
  })

  it('should render ol#activity', () => {
    const vm = new Constructor().$mount()
    expect(vm.$el).toMatchSnapshot()
  })

  it('should render some activity', () => {
    let event = {
      who: '/users/uid',
      what: 'Created a post',
      why: 'unknown',
      when: '2017-12-20T23:01:14.310Z',
      where: '/posts/1'
    }
    const vm = new Constructor({
      propsData: {
        activity: [event]
      }

    }).$mount()
    // expect(cmp.element).toMatchSnapshot()
    expect(vm.$el).toMatchSnapshot()
  })

})
