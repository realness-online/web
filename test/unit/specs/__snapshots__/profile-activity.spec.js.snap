// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile-activity.vue should render ol#activity 1`] = `
<ol
  id="activity"
  itemprop="activity"
/>
`;

exports[`profile-activity.vue should render some activity 1`] = `
<ol
  id="activity"
  itemprop="activity"
>
  <li
    itemscope="itemscope"
    itemtype="/activity"
  >
    <a
      href="/posts/1"
      itemprop="where"
    >
      <b
        itemprop="what"
      >
        Created a post
      </b>
    </a>
     
    <time
      datetime="2017-12-20T23:01:14.310Z"
      itemprop="when"
    />
  </li>
</ol>
`;
